### PR TITLE
Switch HEVC to use internal Core interfaces

### PR DESCRIPTION
**Pull request by artem-shaporenko**
Fixes: #559

### CODE REVIEW

File: _studio/mfx_lib/encode_hw/h265/include/mfx_h265_encode_hw.h [hunk 1]
```diff
@@ -179,12 +178,13 @@ class MFXVideoENCODEH265_HW : public VideoENCODE
     mfxStatus WaitForQueringTask(Task& task);
 
     std::unique_ptr<DriverEncoder>  m_ddi;
-    MFXCoreInterface                m_core;
+    VideoCORE                      *m_core;
```
[the remark this answers is not in the document]
Comment (artem-shaporenko) [reply _studio/mfx_lib/encode_hw/h265/include/mfx_h265_encode_hw.h]: This is good for improvement but let it be separate commit.

**Comment by dvrogozh** [comment]
Guys, why you did not fix commit message _before_ merge?

**Comment by artem-shaporenko** [comment]
What is problem? it was fixed before merge, issue was closed automatically

**Comment by dvrogozh** [comment]
The problem is that we don't have a reference to the fixed issue in the commit message and git history.

**Comment by artem-shaporenko** [comment]
Ok, got it.

**Pull request merged**